x/utils/migutils/migutils.go

```diff
@@ -0,0 +1,57 @@
+package migutils
+
+import (
+	errorsmod "cosmossdk.io/errors"
+	"cosmossdk.io/store/prefix"
+	storetypes "cosmossdk.io/store/types"
+)
+
+// Deletes all keys in the store with the given keyPrefix `maxPageSize` keys at a time
+func SafelyClearWholeMap(store storetypes.KVStore, keyPrefix []byte, maxPageSize uint64) error {
```

Comment:
should this have a test to make sure it works as expected?

Reply:
+1

Reply:
Safer than before for sure tho!